Ftcash Pre-Prod Logs Submitted

gayatriyogesh opened the pull request:
Ftcash Pre-Prod Logs Submitted

sourabhsaini11 commented:
@gayatriyogesh, kindly get below issues cleared
common_issue
- timestamp for /action calls must be greater than the timestamp of past call.
- quote.breakup summation & quote.price value should match to each other
- OTHER_UPFRONT_CHARGES is missing in quote.breakup
- valid id instead of INSTALLMENT_ID_INVOICE_LOAN should be present in payments
- context.timestamp must match format "date-time"
- bpp & bap info from context must remain same across all calls

on_search_1
- INFO tag-group should be present instead of general-info at items

on_search_2
- INFO tag-group should be present instead of general-info at items
- CONSENT_INFO & xinput are missing in items
- tag OFFLINE_CONTRACT is missing at SETTLEMENT_TERMS
- tag-group LSP_INFO is missing

on_search_3
- INFO tag-group should be present instead of general-info at items
- CONSENT_INFO is missing in items[0]
- for INFO tag-group at items[1], tags INSTALLMENT_AMOUNT, OTHER_UPFRONT_CHARGES, NET_DISBURSED_AMOUNT, ANNUAL_PERCENTAGE_RATE, COOL_OFF_PERIOD, REPAYMENT_FREQUENCY, NUMBER_OF_INSTALLMENTS_OF_REPAYMENT, INSURANCE_CHARGES & KFS_LINK are missing
- tag OFFLINE_CONTRACT is missing at SETTLEMENT_TERMS
- tag-group LSP_INFO is missing

on_select
- for INFO tag-group at items, tags INSTALLMENT_AMOUNT, ANNUAL_PERCENTAGE_RATE, COOL_OFF_PERIOD, REPAYMENT_FREQUENCY, NUMBER_OF_INSTALLMENTS_OF_REPAYMENT & KFS_LINK are missing
- multiple CONTACT_INFO tag-groups present (for /on_select_1)
- tag-group LSP_INFO is missing (for /on_select_2)

on_init
- for INFO tag-group at items, tags INSTALLMENT_AMOUNT, ANNUAL_PERCENTAGE_RATE, COOL_OFF_PERIOD, REPAYMENT_FREQUENCY, NUMBER_OF_INSTALLMENTS_OF_REPAYMENT & KFS_LINK are missing
- tag OFFLINE_CONTRACT is missing at SETTLEMENT_TERMS
- payments.id mismatch to what was sent in past call (for /on_init_2)

on_confirm
- payments.id mismatch to what was sent in past call
- for INFO tag-group at items, tags KFS_LINK is missing
- collected_by is sent additionally in INSTALLMENT objects
- tags GRO_DESIGNATION & GRO_ADDRESS are missing in CONTACT_INFO tag-group

on_update
- timestamp & message_id can't be same as timestamp & message_id of past call
- file contains payload of /on_status call
- for INFO tag-group at items, tags KFS_LINK is missing
- collected_by is sent additionally in INSTALLMENT objects
- tags GRO_DESIGNATION & GRO_ADDRESS are missing in CONTACT_INFO tag-group

Flow-2
on_update_1
- time.label is FORECLOSURE while flow is for PRE_PART_PAYMENT
- for INFO tag-group at items, tags KFS_LINK is missing

on_update_2
- bpp & bap info are not same as to what was being sent earlier

Flow-3
on_update_1
- time.label is FORECLOSURE while flow is for MISSED_EMI_PAYMENT
- status should be DELAYED instead of NOT-PAID

on_update_2
- bpp & bap info are not same as to what was being sent earlier
- status should be DEFERRED instead of PAID

Flow-4
on_update_2
- bpp & bap info are not same as to what was being sent earlier
- status should be DEFERRED instead of PAID

Flow-5
on_update_2
- file is empty


IGM
issue(close)
- org.name should be equal to bap_id::domain

on_issue
- Timestamp should be in date-time format
- 'created_at' should match issue's 'created_at'
- org.name should be equal to bpp_id::domain

on_issue_status
- 'created_at' should match issue's 'created_at'
- created_at, updated_at, timestamp should be in date-time format
- org.name should be equal to bpp_id::domain
- Message ID mismatch in /issue_status and /on_issue_status

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check its

gayatriyogesh commented:
Hi @sourabhsaini11  I have fixed all issues and updated logs. please check its

1) **tag INVOICE_NUMBER value should be a valid value** - The inum field is the correct field for invoice number. It's just that the value in the test set looked like a price value so this false positive comment was raised .this is the inum value in the json received from GSP

2) **CONSENT_INFO tag-group is missing in items[0]** -its should not be there its a Invoice Based Loans without Account Aggregator & monitoring consent flow

sourabhsaini11 commented:
@gayatriyogesh, please fix below issues
common_issue
- price summation for quote.breakup should match quote.price.value (value must be precise upto 2 decimal places)

on_search_2
- tag KFS_LINK is missing in INFO tag-group(for item obj with parent_item_id)

on_select
- tag REPAYMENT_FREQUENCY value must be in P1M format
- LSP_INFO tag-group is missing(on_select_1 only)
- title OTHER_UPFRONT_CHARGES is missing in quote.breakup(on_select_1 only)

init_3
- tag DELAY_INTEREST is missing in SETTLEMENT_TERMS tag-group

on_init_3
- incorrect JSON at 
```json
 },
                        {
                           "descriptor":{
                              "code":"COOL_OFF_PERIOD",
                              "name":"cool off period",
                              "short_desc":"Cooling off/look-up period during which borrower shall not be charged any penalty"
                           },
                           "value":"P30D"
                        },f
                        {
```

init_4
- tag DELAY_INTEREST is missing in SETTLEMENT_TERMS tag-group
- message_id same as past /init_3 call

on_init_4
- collected_by is sent additionally 
- message_id mismatch between /init_4 & /on_init_4

conifrm
- tag DELAY_INTEREST is missing in SETTLEMENT_TERMS tag-group
- payments.params is changed between /init & /conifrm call

on_status
- tags GRO_DESIGNATION & GRO_ADDRESS are missing in CONTACT_INFO tag-group
- title OTHER_UPFRONT_CHARGES is missing in quote.breakup

on_status_unsolicated
- message_id same as /confirm call

IGM
on_issue_status
- respondent_actions.updated_at must match format \"date-time\

Flows: 2-5 are clone of ICICI logs

Flow-2
on_update
- items & fulfillments objects mismatch between /on_confirm & /on_update

on_update_2
- payments[0].params.amount & tag SETTLEMENT_AMOUNT value mismatch between /on_update_1 & /on_update_2

Flow-3
on_update
- items & fulfillments objects mismatch between /on_confirm & /on_update

on_update_2
- bap_uri mismatch between /on_update_1 & /on_update_2

Flow-4, 5
on_update
- items & fulfillments objects mismatch between /on_confirm & /on_update

on_update_2
- payments[0].params.amount, payments[2].params.amount & tag SETTLEMENT_AMOUNT value mismatch between /on_update_1 & /on_update_2
- upon FORECLOSURE, code should change to COMPLETED at fulfillments.state

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check its

We have recently getting some observations in the pre-production logs concerning buyer payloads. Upon discussing the matter with buyer, I was informed that their team is currently occupied with other high-priority tasks, and they will be able to address the issues after a few days.

Given this, for the time being, can we proceed by temporarily ignoring the buyer-side issues

sourabhsaini11 commented:
@gayatriyogesh, please get the below issues fixed
on_confirm
- cancellation_terms contains both fulfillment_state as SANCTIONED, DISBURSED is missing

on_status, on_update
- cancellation_terms contains both fulfillment_state as DISBURSED, SANCTIONED is missing

on_update
- parent_item_id is missing
- tag-group name should be INFO instead of LOAN_INFO, also tags present are also incorrect
- CONTACT_INFO tag-group mismatch betwenn /on_update & past cancellation_terms

Flow-4, 5
on_update_1
- descriptorcode should not be COMPLETED at fulfillments.fulfillment_state

on_update_2
- status should change to DEFERRED instead of PAID for payment objects having label as INSTALLMENT

Antriksh-Parmar commented:
@sourabhsaini11 We are not using DEFERRED
cc: @92shreyansh

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check its

sourabhsaini11 commented:
@gayatriyogesh, please resolve below issues
Remove Jtechnoparks, dmi-ondcpreprod changes present in commit

common_issue
- quote.breakup summation & quote.price value should match to each other

on_select
- head.descriptor.name should change on the basis of current index of headings array
- index.cur value should also inc. on each on_select call

on_init
- index.max should be 3
- head.descriptor.name should change on the basis of current index of headings array
- index.cur value should also inc. on each on_init call

Flow-2
on_update
- PRE_PART_PAYMENT amount mismatch between /update(1000) & /on_update(1735)

Flow-3
on_update
- additional titles present in quote.breakup
- tile LATE_FEE_AMOUNT is missing in quote.breakup
- status should change to DEFERRED at payments[2](/on_update_2 only)

Flow-4, 5
on_update_2
- additional title PRE_PAYMENT_CHARGE present in quote.breakup
- tile FORCLOSUER_CHARGES is missing in quote.breakup
- status should change to DEFERRED at payments[2](/on_update_2 only)
- tag OFFLINE_CONTRACT is missing in SETTLEMENT_TERMS tag-group

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check its.
1) status should change to DEFERRED at payments[2](/on_update_2 only)-We are not using DEFERRED
cc: @92shreyansh

sourabhsaini11 commented:
@gayatriyogesh, please fix below issues
common_issue
- tag OFFLINE_CONTRACT is missing(on_update)
- quote.breakup summation & quote.pice value should match to each other

on_init
- head.descriptor.name should change on the basis of current index of headings array

on_select
- head.descriptor.name should change on the basis of current index of headings array

Flow 1
Init_2
- message.order.payments[0].tags.descriptor_code.SETTLEMENT_TYPE and SETTLEMENT_BASIS is missing in the payload.

sourabhsaini11 commented:
@gayatriyogesh according to flow-Invoice Based Loans with Account Aggregator & monitoring consent, there has to be 3 search-on_search api pair calls. Please confirm once if you are targeting wiithout Account Aggregator & monitoring consent

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check its.

 - we are targeting without Account Aggregator & monitoring consent.
 
 -Init_2-message.order.payments[0].tags.descriptor_code.SETTLEMENT_TYPE and SETTLEMENT_BASIS is missing in the payload.-> please ignore buyer side points

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check its.

sourabhsaini11 commented:
@gayatriyogesh as discussed with @92shreyansh please upload fresh transactional logs.

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check it.

I requested you if possible,please  given all the issues at one time.
cc:  @Antriksh-Parmar  @92shreyansh

sourabhsaini11 commented:
@gayatriyogesh 
on_search_2
- parent_item_id should match with item.id provided in on_search_1 at items[1]
- summation for tags PRINCIPAL, INTEREST, PROCESSING_FEE should amount to item.price, if not then discrepancies b/w item.price is present at items[1] in INFO tag-group

on_select
- item.price mismatch b/w on_search_2 & on_select

common_issues
- At quote, price.value must be precise upto 2 decimal places
- virtual_payment_address shouldn't be present if SETTLEMENT_TYPE is not UPI

on_init_4
- cancellation_terms array is changed b/w on_init_3 & on_init_4

on_confirm
- cancellation_terms array is changed b/w on_init_4 & on_confirm

PRE_PART_PAYMENT
on_update
- cancellation_terms array is changed b/w on_update & on_confirm
- for NOT-PAID installments, their status should change to DEFERRED instead of PAID (on_update_2 only)

MISSED_EMI_PAYMENT
on_update_1
- status should be DELAYED instead of NOT-PAID
- cancellation_terms array is changed b/w on_update & on_confirm

on_update_2
- cancellation_terms array is changed b/w on_update & on_confirm
- why is bap_id & bap_uri same
- status should be DEFERRED instead of PAID at payments[2]

FORECLOSURE
on_update
- cancellation_terms array is changed b/w on_update & on_confirm
- item.price mismatch b/w on_update_1 & on_confirm

on_update_2
- cancellation_terms array is changed b/w on_update & on_confirm
- why is bap_id & bap_uri same
- item.price mismatch b/w on_update_1 & on_update_2
- status should be DEFERRED instead of PAID at payments[2]

amitsinha07 commented:
@gayatriyogesh 
# Feedbacks for IGM
## **Issue_CLOSE**
- `Created_at`: Does not match with the issue's created_at.
- `message/issue/issue_actions/respondent_actions/1/updated_at`: Should be less than or equal to `issue/updated`.

## **On_issue**
- `TTL`: Must be `PT30S` as per the API contract.
- All timestamps are matching with the issue call, which is not valid.

## **Issue_status**
- `TTL`: Must be `PT30S` as per the API contract.

## **On_issue_status**
- `TTL`: Must be `PT30S` as per the API contract.
- `message/issue/resolution_provider`: Should not be present as per the API contract.
- `message/issue/issue_actions/respondent_actions[0]/updated_at`: Should match with `on_issues`'s `message/issue/issue_actions/respondent_actions[0]/updated_at`.
- All timestamps are matching with the previous call, which is not valid.

gayatriyogesh commented:
Hi @sourabhsaini11 I have fixed all issues and updated logs. please check it.
1.for NOT-PAID installments, their status should change to DEFERRED instead of PAID (on_update_2 only)-PRE_PART_PAYMENT
2.status should be DEFERRED instead of PAID at payments[2]-MISSED_EMI_PAYMENT
3.status should be DEFERRED instead of PAID at payments[2]-FORECLOSURE

"We are not using DEFERRED" please note this one 
cc: @92shreyansh @Antriksh-Parmar 


4.virtual_payment_address shouldn't be present if SETTLEMENT_TYPE is not UPI - this is from the buyer side Pramaan


In IGM
5. message/issue/resolution_provider: Should not be present as per the API contract.-On_issue_status

@amitsinha07 I am not able to find these changes in the ondc portal 

6. Issue_CLOSE- below is from the buyer side Pramaan
       1.Created_at: Does not match with the issue's created_at.
      2.  message/issue/issue_actions/respondent_actions/1/updated_at: Should be less than or equal to issue/updated.


please let me know what I should do

sourabhsaini11 commented:
@gayatriyogesh 
on_search_2
- parent_item_id at items[1] should also match with item.id provided at items[0], in short the selected item object which was sent in on_search_1 should remain same in on_search_2 as well

common_issue
- INTEREST is assigned as 525.7 at on_search_2 but value used is 525 at quote.breakup, should be precise upto 2 decimal places as well

PRE_PART_PAYMENT
on_update_2
- if you won't be changing it INSTALLMENT status to DEFERRED state then atleast change it to PAID one

MISSED_EMI_PAYMENT
on_update_1
- payments[0].status should be NOT-PAID.

amitsinha07 commented:
@gayatriyogesh

## **Issue**
**TTL Validation**: 
  - TTL = PT30S as per the API Contract.
**ID Format Validation**: 
  - message/issue/id should be in "uuid" format.


## **Issue_Close**
**TTL Validation**: 
  - TTL = PT30S as per the API Contract.


## **On_Issue**.

**Timestamp Validation**: 
  - message/issue/issue_action/respondent_actions/index[0]/updated_at should be less than or equal to message/issue/update_at in the "on_issue" call.
  - issue/created_at and issue/updated_at cannot be same


## **Issue_Status**
**TTL Validation**: 
  - TTL = PT30S as per the API Contract.


## **On_Issue_Status**
**Timestamp Mismatch**: 
  - created_at is mismatched with the created_at in the "issue" call.
  - message/issue/issue_actions/respondent_actions/0/updated_at should be the same as or less than message/issue/update_at.
**Invalid Resolution Provider**: 
  - message/issue/resolution_provider should not be present if the respondent action "Resolved" is absent.
**Domain Validation**: 
  - Domain in message/issue/issue_actions/respondent_actions/[0] does not match the domain in the context.
**Invalid Timestamps**: 
  - message/issue/created_at format is wrong.
  - message/issue/updated_at format is wrong.
message/issue/resolution/actions- triggered object is missing.

gayatriyogesh commented:
Hi @sourabhsaini11 

We have updated the Flow 1,2 3,4 ,5 Logs as per the feedback

gayatriyogesh commented:
Hi @amitsinha07 
We have updated the Flow IGM Logs as per the feedback, please I request that to you ignore Pramaan side logs (buyer side apis call)

sourabhsaini11 commented:
@gayatriyogesh 
common_issue
- quote.breakup summation should match quote.price with 2 decimal precision

search_2
- items.id should match with one of item id's provided in prev call

confirm
- form_response.submission_id shouldn't be similar to what was sent in init_4

Flows-2, 3, 4 & 5
unsolicated on_update_2
- transaction_id & message_id can't be same

Provide unsolicated on_status calls as well sent by BPP for application/html forms

amitsinha07 commented:
@gayatriyogesh f

## Feedbacks for IGM

### issue_status
  - `context/ttl`: Should be `"PT30S"` as per the API Contract but is currently `"PT30M"`.

### on_issue_status
- **Validation Rules**:
  - `message/issue/issue_actions/respondent_actions/0/respondent_action`: Must include the value `"PROCESSING"` as it is from `on_issue`.

gayatriyogesh commented:
Hi @amitsinha07 
We have updated the Logs as per the feedback

sourabhsaini11 commented:
@gayatriyogesh 
on_search_2
- parent_item_id at items[1] should also match with item.id provided at items[0], in short the selected item object which was sent in on_search_1 should remain same in on_search_2 as well
- incorrect format for tag OFFER_VALIDITY, should be P15D instead 

common_issue
- quote.breakup summation should match quote.price with 2 decimal precision

on_init_4
- invalid JSON format

FORECLOSURE
on_update_1
- How can fulfillments.state be marked as COMPLETED

PRE_PART_PAYMENT
on_update_2
- if you won't be changing it INSTALLMENT status to DEFERRED state then atleast change it to PAID one

sourabhsaini11 commented:
@gayatriyogesh, let's connect tomorrow. In the meantime you can shoot your queries in this thread. We will resolve it here.

sourabhsaini11 commented:
@gayatriyogesh 
As part of on_search_2 items array will have at least 2 elements one is the base item id that was selected in on_search 1 which was selected and there are multiple items each having a parent_item_id comprising of the value of the item_id in on_search_1

gayatriyogesh commented:
@sourabhsaini11 logs are updated please check it

gayatriyogesh commented:
@sourabhsaini11 @amitsinha07 can you give feedback here for the last logs submission

cc @Antriksh-Parmar @92shreyansh

amitsinha07 commented:
@gayatriyogesh 

## On_issue_status
- **Message/issue/resolution_provider**: Should not be present as `respondent_action` is not "Resolved".  
- **Message/issue/issue_actions/respondent_actions/0/updated_at**: Mismatch with the corresponding field in the `on_issue` call.  
- **Message/issue/issue_actions/respondent_actions/0/updated_by/contact/email**: Mismatch with the corresponding field in the `on_issue` call.  
- **Message/issue/issue_actions/respondent_actions/0/Short_desc**: Mismatch with the corresponding field in the `on_issue` call.

gayatriyogesh commented:
Hi @amitsinha07 
We have updated the Flow IGM Logs as per the feedback,

amitsinha07 commented:
@gayatriyogesh 
## On_issue_status

The value `"Processing"` is missing for the following path:  
`message/issue/issue_actions/respondent_actions/0/respondent_action`.

gayatriyogesh commented:
Hi @amitsinha07
We have updated the Flow IGM Logs as per the feedback,

amitsinha07 commented:
@gayatriyogesh 

Your logs are Cleared for IGM 1.0

Antriksh-Parmar commented:
@amitsinha07 @92shreyansh anything on core APIs?

gayatriyogesh commented:
Thanks @Antriksh-Parmar @92shreyansh @sourabhsaini11 @amitsinha07
